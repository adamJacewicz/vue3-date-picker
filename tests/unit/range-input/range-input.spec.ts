import { mount, VueWrapper } from '@vue/test-utils';
import RangeInput            from '@/components/molecules/date-picker/components/range-input.vue';
import CInput                from '@/components/atoms/input.vue';
import dayjs                 from 'dayjs';

describe('RangeInput', () => {
  let wrapper!: VueWrapper<any>;

  beforeEach(() => {
    wrapper = mount(RangeInput, {
      props: {
        placeholder: [ 'Check in', 'Check out' ],
        format: 'YYYY/MM/DD',
        label: 'Label',
        disabled: false,
        modelValue: [ dayjs(new Date('2021-03-02')), dayjs(new Date('2021-04-02')) ],
      },
    });
  });

  it('should render component with correct class', () => {
    expect(wrapper.classes()).toContain('range-input');
  });

  it('should render formatted dates', () => {
    const [ startInput, endInput ] = wrapper.findAllComponents(CInput);
    const [ startDate, endDate ] = wrapper.vm.formattedValue;
    expect(startInput.props('value')).toEqual(startDate);
    expect(endInput.props('value')).toEqual(endDate);
  });

  it('render clear button', () => {
    const button = wrapper.get('.range-input__clear-btn');
    expect(button.isVisible()).toBe(true);
  });


  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
