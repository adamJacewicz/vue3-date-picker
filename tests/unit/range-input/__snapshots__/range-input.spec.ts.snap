// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeInput should match snapshot 1`] = `
<div class="range-input"><label class="c-input__label" disabled="false" placeholder="Check in"> <input placeholder="Check in" class="c-input"></label><span class="range-input__separator">â†’</span><label class="c-input__label" disabled="false" placeholder="Check out"> <input placeholder="Check out" class="c-input"></label><button aria-label="clear-button" class="range-input__clear-btn c-button range-input__clear-btn"><svg width="24" height="24" viewBox="0 0 48 48" fill="none" xmlns="http://www.w3.org/2000/svg">
      <rect width="48" height="48" fill="white" fill-opacity="0.01"></rect>
      <path d="M24 44C35.0457 44 44 35.0457 44 24C44 12.9543 35.0457 4 24 4C12.9543 4 4 12.9543 4 24C4 35.0457 12.9543 44 24 44Z" fill="none" stroke="#333" stroke-width="2" stroke-linejoin="miter"></path>
      <path d="M29.6569 18.3431L18.3432 29.6568" stroke="#333" stroke-width="2" stroke-linecap="round" stroke-linejoin="miter"></path>
      <path d="M18.3432 18.3431L29.6569 29.6568" stroke="#333" stroke-width="2" stroke-linecap="round" stroke-linejoin="miter"></path>
    </svg></button><svg class="icon" xmlns="http://www.w3.org/2000/svg" stroke="currentColor" width="48px" height="48px" viewBox="0 0 512 512" style="display: none;">
    <rect fill="none" stroke-linejoin="round" stroke-width="24" x="48" y="80" width="416" height="384" rx="48"></rect>
    <path fill="none" stroke-linejoin="round" stroke-width="24" stroke-linecap="round" d="M128 58v32M384 58v32M464 160H48"></path><text x="256" fill="currentColor" y="400" stroke="none" font-size="256px">22</text>
  </svg></div>
`;
